src/main/java/gregtech/api/capability/impl/MultiblockRecipeLogic.java

```diff
@@ -103,6 +103,25 @@ protected IMultipleTankHandler getOutputTank() {
         return controller.getOutputFluidInventory();
     }
 
+    @Override
+    protected boolean canWorkWithInputs() {
+        MultiblockWithDisplayBase controller = (MultiblockWithDisplayBase) metaTileEntity;
+        if (controller instanceof RecipeMapMultiblockController) {
+            RecipeMapMultiblockController distinctController = (RecipeMapMultiblockController) controller;
+
+            if (distinctController.canBeDistinct() && distinctController.isDistinct()) {
+                if (invalidatedInputList.isEmpty() || !metaTileEntity.getNotifiedFluidInputList().isEmpty()) return true;
```

Comment:
Should this check `invalidatedInputList.isEmpty() && !metaTileEntity.getNotifiedItemInputList.isEmpty() ||...` instead of just checking if we have no invalidated any inputs previously?

Reply:
we only care if the inputs changed if it is currently invalidated